src/main/java/seedu/address/model/task/Task.java

```diff
@@ -273,4 +273,17 @@ public void markAsDone() {
         return (Task task) -> new Observable[] { task.getDone(),task.getDateString()};
     }
 
+	public boolean editDetail(String type, String details) throws IllegalValueException {
+		switch(type) {
+    	case "name": this.name = new Name(details); break;
+//    	case "tag": this.tag = new UniqueTagList(details); break;
+    	case "recurring": this.recurring = new Recurring(details); break;
+    	case "startDate": this.date = new EventDate(details,((EventDate) date).getEndDate().substring(0, 10)); break;
```

Comment:
If the old date is not an EventDate, there will have ClassCastException. So maybe it is better to check the type of the old date?